### PR TITLE
Make query items return iterator[any] rather than items

### PR DESCRIPTION
**Pull request by johanste**
Any projections or aggregate queries may not return an iterator[Item] but rather an iterator of either scalar (for VALUE aggregate queries) or arbitrary documents (for projections).

This change makes this clear and it also squirrels away the metadata/header result of the query onto the iterator rather than on each item returned.

### CODE REVIEW

File: azure/cosmos/__init__.py [hunk 1]
```diff
@@ -495,7 +496,8 @@ def query_items(
             partition_key=partition_key,
         )
         headers = self.client_context.last_response_headers
-        yield from [Item(headers, item) for item in items]
+        items_iteratable = iter(items)
```
Comment (mmacy) [inline azure/cosmos/__init__.py]: `items_iteratable` > `items_iterable`

File: azure/cosmos/query_iterator.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import collections.abc
+
+
+class QueryResultIteratable(collections.abc.Iterator):
```
Comment (mmacy) [inline azure/cosmos/query_iterator.py]: `QueryResultIteratable` > `QueryResultIterable`

**Review by mmacy — CHANGES_REQUESTED** [review]
Spelling on *iterable, otherwise LGTM.

**Comment by mmacy** [comment]
@johanste Oh, and do any examples.py or README snippets need updating along with this change?

File: azure/cosmos/__init__.py (hunk 1 above)
Comment (johanste) [reply azure/cosmos/__init__.py]: Changing to iterator.

File: azure/cosmos/query_iterator.py (hunk 2 above)
Comment (johanste) [reply azure/cosmos/query_iterator.py]: Changing to iterator

**Comment by johanste** [comment]
Uber-question - what should we call an instance of something that was returned by query_items. It isn't necessarily an item. Is it a result? Or a document? Or something else?

File: README.md [hunk 3]
```diff
@@ -139,8 +139,8 @@ This example queries a container for items with a specific `id`:
 database = client.get_database(test_database_name)
 container = database.get_container(test_container_name)
 
-# Get a list of items from the container
-items = list(container.query_items(query='SELECT * FROM root r WHERE r.id="something"'))
+# Get a list of results from the container
```
Comment (johanste) [inline README.md]: @mmacy, what term should we use for the things returned by the API? Thoughts?

Comment (mmacy) [reply README.md]: @johanste I think result is fine, as long as we add type hints for what those types might be in the `:returns:` doc for `query_items`.

Comment (mmacy) [reply README.md]: Though, you've got those docs in `QueryResultIterator`, so perhaps that's sufficient.

**Review by mmacy — APPROVED** [review]

File: azure/cosmos/__init__.py [hunk 4]
```diff
@@ -33,6 +37,45 @@ class User:
     pass
 
 
+class PartitionKey(dict):
+    """ Key used to partition a container into logical partitions.
+
+    See https://docs.microsoft.com/en-us/azure/cosmos-db/partitioning-overview#choose-partitionkey for more information
```
Comment (mmacy) [inline azure/cosmos/__init__.py]: Minor: URLs should not include locale identifiers (`en-us`) so browsers can be redirected automatically to the localized versions of articles.

Comment (johanste) [reply azure/cosmos/__init__.py]: Updated (removed local identifier)

**Pull request merged**